### PR TITLE
parameter validations

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**
## Type of PR

- [X ] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Is to define a parameter validation specification to be used in C#, java, Node versions of helium application

## Issues Closed or Referenced

- References #243

### CODE REVIEW

File: docs/ParameterValidation.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    - Response Body: Invalid actorid parameter
+
+- Name: genre
+- Type: string
+- Parameter Validation:
+  - Valid input: genre length [3, 20]
+    - Example: 'War' , 'Documentary'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by genre as [{}, {},..], empty array when no results,  where movie.genres contains genre
```
Comment (bartr) [inline docs/ParameterValidation.md]: This is currently case insensitive - should we call that out?

File: docs/ParameterValidation.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+- Fail on first error
+
+## Movies
+
+### API /api/movies
+
+### Movies Query Parameters
+
+- Name: query (search)
```
Comment (bartr) [inline docs/ParameterValidation.md]: should we call out case insensitive?

File: docs/ParameterValidation.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+- Name: pageNumber
+- Type: integer
+- Description: pageNumber sets the first position to return from the results of the query
+- Parameter Validation:
+  - Default value: 1
+  - Valid input range: [1, 10000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies based on pageNumber as [{}, {},..], empty array when no results
+  - Invalid input - input that does not parse
```
Comment (bartr) [inline docs/ParameterValidation.md]: or is out of range

File: docs/ParameterValidation.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    - Example 'tt0114746'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: Single movie document as JSON
+  - Invalid input: movieId not conforming to well formed movieId rule
+    - Example:'tt1234', 'nm1234'
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid movieId parameter
+  - Invalid input: movieId does not exist
```
Comment (bartr) [inline docs/ParameterValidation.md]: This isn't really invalid input - it's not found - should it be Valid input: movieId does not exist instead?

File: docs/ParameterValidation.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    - Example: 'nm1265067'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: Single actor document as JSON
+  - Invalid input: actorId not conforming to well formed movieid rule
+    - Example: 'nm1234'
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid actorId parameter
+  - Invalid input: actorId does not exist
```
Comment (bartr) [inline docs/ParameterValidation.md]: same Valid input comment as movies

File: docs/ParameterValidation.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies based on pageNumber as [{}, {},..], empty array when no results
+  - Invalid input: input that does not parse or is out of range
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageNumber parameter
+
+### API Parameters
+
```
Comment (aflinchb) [inline docs/ParameterValidation.md]: Name this header: "Movies API Parameters" to be consistent with Actors header below

File: docs/ParameterValidation.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid year parameter
+
+- Name: rating
+- Type: double
+- Parameter Validation:
+  - Valid input range: [0.0, 10.0]
+  - Round off more than single decimal to a single decimal
+    - Example: round off 7.253896 to 7.3
```
Comment (jofultz) [inline docs/ParameterValidation.md]: When we talked I think we said that we'd take whatever the round function for the language did by default.  However, here it sounds like we will always round up at 5.  I think that is okay and gives good consistency, but let's call it out if it is a rule.  Round to a single digit and round up at 5.

File: docs/ParameterValidation.md (hunk 3 above)
Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: Added the "out of range"

File: docs/ParameterValidation.md (hunk 5 above)
Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: Updated!

File: docs/ParameterValidation.md (hunk 7 above)
Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: updated with "based on round function for the language"

Comment (bartr) [reply docs/ParameterValidation.md]: I don't see the update when I review in github web ... what am I missing?

File: docs/ParameterValidation.md (hunk 1 above)
Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: this is addressed

**Pull request merged**